src/transaction.js

```diff
@@ -250,8 +399,22 @@ Transaction.prototype.getId = function () {
   return [].reverse.call(this.getHash()).toString('hex')
 }
 
-Transaction.prototype.toBuffer = function () {
-  var buffer = new Buffer(this.byteLength())
+Transaction.prototype.toBuffer = function (withWitness) {
```

Comment:
It was mentioned yesterday, but worth documenting: a param or separate serialization methods are required because you *should* be able to convert a segwit tx into the older version. The other way around (old to new) doesn't make sense, but is still technically valid to witness-enabled software.